src/Pelagos/Bundle/AppBundle/Resources/public/js/download.js

```diff
@@ -4,42 +4,129 @@ $(document).ready(function()
 
 function startDownload(id)
 {
-    if ($("#download_splash").length) {
-        return;
-    }
-
     $('<div id="download_splash" />').appendTo("#pelagos-content").load(
-        Routing.generate("pelagos_app_download_default", {"id": id}),
-        function () {
-            initializeDownload(id);
-        }
+        $.getJSON( Routing.generate("pelagos_app_download_default", {"id": id}), function( data ) {
+            vex.dialog.open({
+                className: "vex-theme-os",
+                unsafeMessage: getHtmlForDownload(data),
+                buttons: [
+                    $.extend({}, vex.dialog.buttons.YES, { text: "Download", click: function($vexContent, event) {
+                            if (!data.remotelyHosted) {
+                                $.getJSON(Routing.generate("pelagos_app_download_http", {"id": id}), function (data) {
+                                }).done(function (data) {
+                                    window.location = `${data.downloadUrl}`;
+                                });
+                            }
+                        }}),
+                    $.extend({}, vex.dialog.buttons.NO, { text: "Cancel" })
+                ]
+            });
+            if (data.remotelyHosted) {
+                $(":button[type='submit']").remove();
```

Comment:
Maybe it is wise to be a little more specific on the button if possible, or else any additional submit buttons will also be removed.